tests/TestCase/Http/ClientTest.php

```diff
@@ -1157,4 +1158,41 @@ public function testAddMockResponseCustomInvalidDecision(): void
 
         $client->post('http://example.com/path');
     }
+
+    /**
+     * Check that auth adapters can be used without a set App.namespace
+     * config value.
+     */
+    public function testNoNameSpace()
```

Comment:
Could have saved yourself quite some effort by just adding a test for `App::className()` :)

Reply:
This isn't exactly a unit test for App::className() either.

Reply:
> This isn't exactly a unit test for App::className() either.

I have added one. Given that the root cause of the problem is now tested for I don't think this test is not necessary, it just adds bloat.

Reply:
I wanted to replicate my exact issue, which wasn't a lot of effort, as this was a copy of a different test.
Anyway, I agree that testing the actual method makes more sense, so I removed my test again.